scripts/generate_linkml_docs.py

```diff
@@ -1,26 +1,64 @@
 #!/usr/bin/env python3
 """Script to generate linkml markdown documents"""
 
-from typing import TextIO
 from pathlib import Path
-
+import sys
+import subprocess
+from tempfile import TemporaryDirectory
 from typer import run
 from linkml.generators.markdowngen import MarkdownGenerator
+from script_utils.cli import echo_failure, echo_success
 
 HERE = Path(__file__).parent.resolve()
+LINKML_YAML = HERE.parent / "src" / "schema" / "submission_centric_schema.yaml"
 DOCS_DIR = HERE.parent / "docs" / "schema_markdown"
 
 
-def generate_linkml_markdown(linkml_yaml: TextIO):
+def generate_linkml_markdown(docs_dir: Path):
     """Function to generate markdown documentations of a given schema"""
-    return MarkdownGenerator(schema=linkml_yaml, directory_output=True).serialize(
-        directory=DOCS_DIR
+    return MarkdownGenerator(schema=str(LINKML_YAML), directory_output=True).serialize(
+        directory=docs_dir
     )
 
 
-def main(linkml_yaml):
-    """Main function"""
-    generate_linkml_markdown(linkml_yaml)
+class ContentDifference(Exception):
+    """Custom exception to raise non-equality of the compared folders"""
+
+    pass
+
+
+def compare_folders(expected: Path, observed: Path):
+    """Function to check equality of contents of two folders"""
+
+    with subprocess.Popen(
+        ["diff", "-arq", str(expected), str(observed)],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding="utf-8",
+    ) as process:
+        stdout, _ = process.communicate()
+
+        if process.returncode != 0:
+            raise ContentDifference(f"Documentations are outdated. {stdout}")
```

Comment:
Have you considered [dircmp](https://docs.python.org/3/library/filecmp.html) instead? Relying on a particular implementation of `diff`, GNU diff in this case, can be error prone. Especially since this is not going to be run in our devcontainers but on 3rd party CI VMs.